From 35359c1b34f1f23251dd59f1d760fe3a35c73e99 Mon Sep 17 00:00:00 2001
From: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Date: Mon, 3 Mar 2014 20:51:03 -0800
Subject: ASoC: rsnd: nothing to do on rsnd_dai_remove()

rsnd_dai_remove() called rsnd_path_exit(),
but these memory will be cleaned automatically.
Because it is created by devm_kzalloc().
nothing to do on rsnd_dai_remove()

Signed-off-by: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Signed-off-by: Mark Brown <broonie@linaro.org>
(cherry picked from commit 4076220767416b8b65009dd57eeb317c38d41baa)
Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
---
 sound/soc/sh/rcar/core.c | 36 ------------------------------------
 1 file changed, 36 deletions(-)

diff --git a/sound/soc/sh/rcar/core.c b/sound/soc/sh/rcar/core.c
index e8e585de7251..92e155b7b6a7 100644
--- a/sound/soc/sh/rcar/core.c
+++ b/sound/soc/sh/rcar/core.c
@@ -348,14 +348,6 @@ static int rsnd_dai_connect(struct rsnd_mod *mod,
 	return 0;
 }
 
-static int rsnd_dai_disconnect(struct rsnd_mod *mod, struct rsnd_dai_stream *io)
-{
-	io->mod[mod->type] = NULL;
-	mod->io = NULL;
-
-	return 0;
-}
-
 int rsnd_dai_id(struct rsnd_priv *priv, struct rsnd_dai *rdai)
 {
 	int id = rdai - priv->rdai;
@@ -628,26 +620,6 @@ static int rsnd_path_init(struct rsnd_priv *priv,
 	return ret;
 }
 
-static int rsnd_path_exit(struct rsnd_priv *priv,
-			  struct rsnd_dai *rdai,
-			  struct rsnd_dai_stream *io)
-{
-	struct rsnd_mod *mod;
-	int ret = 0, i;
-
-	/*
-	 * remove all mod from rdai
-	 */
-	for (i = 0; i < RSND_MOD_MAX; i++) {
-		mod = io->mod[i];
-		if (!mod)
-			continue;
-		ret |= rsnd_dai_disconnect(mod, io);
-	}
-
-	return ret;
-}
-
 static int rsnd_dai_probe(struct platform_device *pdev,
 			  struct rsnd_priv *priv)
 {
@@ -739,14 +711,6 @@ static int rsnd_dai_probe(struct platform_device *pdev,
 static void rsnd_dai_remove(struct platform_device *pdev,
 			  struct rsnd_priv *priv)
 {
-	struct rsnd_dai *rdai;
-	int i;
-
-	for (i = 0; i < rsnd_rdai_nr(priv); i++) {
-		rdai = rsnd_dai_get(priv, i);
-		rsnd_path_exit(priv, rdai, &rdai->playback);
-		rsnd_path_exit(priv, rdai, &rdai->capture);
-	}
 }
 
 /*
-- 
2.1.2

